src/core/build_target.go

```diff
@@ -92,63 +114,11 @@ type BuildTarget struct {
-	// Shell command to run for test targets.
-	TestCommand string `name:"test_cmd"`
-	// Per-configuration test commands to run.
-	TestCommands map[string]string `name:"test_cmd"`
-	// Represents the state of this build target (see below)
-	state int32 `print:"false"`
-	// True if this target is a binary (ie. runnable, will appear in plz-out/bin)
-	IsBinary bool `name:"binary"`
-	// True if this target is a test
-	IsTest bool `name:"test"`
-	// Indicates that the target can only be depended on by tests or other rules with this set.
-	// Used to restrict non-deployable code and also affects coverage detection.
-	TestOnly bool `name:"test_only"`
-	// True if the build action is sandboxed.
-	Sandbox bool
-	// True if the test action is sandboxed.
-	TestSandbox bool `name:"test_sandbox"`
-	// True if the target is a test and has no output file.
-	// Default is false, meaning all tests must produce test.results as output.
-	NoTestOutput bool `name:"no_test_output"`
-	// True if this target needs access to its transitive dependencies to build.
-	// This would be false for most 'normal' genrules but true for eg. compiler steps
-	// that need to build in everything.
-	NeedsTransitiveDependencies bool `name:"needs_transitive_deps"`
-	// True if this target blocks recursive exploring for transitive dependencies.
-	// This is typically false for _library rules which aren't complete, and true
-	// for _binary rules which normally are, and genrules where you don't care about
-	// the inputs, only whatever they were turned into.
-	OutputIsComplete bool `name:"output_is_complete"`
-	// If true, the rule is given an env var at build time that contains the hash of its
-	// transitive dependencies, which can be used to identify the output in a predictable way.
-	Stamp bool
-	// If true, the target must be run locally (i.e. is not compatible with remote execution).
-	Local bool
-	// If true, the executed commands will exit whenever an error is encountered (i.e. shells
-	// are executed with -e).
-	ExitOnError bool
-	// If true, the target is needed for a subinclude and therefore we will have to make sure its
-	// outputs are available locally when built.
-	NeededForSubinclude bool `print:"false"`
-	// Marks the target as a filegroup.
-	IsFilegroup bool `print:"false"`
-	// Marks the target as a remote_file.
-	IsRemoteFile bool `print:"false"`
-	// Marks the target as a text_file.
-	IsTextFile bool `print:"false"`
-	// Marks that the target was added in a post-build function.
-	AddedPostBuild bool `print:"false"`
-	// If true, the interactive progress display will try to infer the target's progress
-	// via some heuristics on its output.
-	ShowProgress bool `name:"progress"`
+	Test     *TestFields       `name:"test"`
 	// If ShowProgress is true, this is used to store the current progress of the target.
 	Progress float32 `print:"false"`
 	// The results of this test target, if it is one.
 	Results TestSuite `print:"false"`
```

Comment:
Can you move the Results field into the sub-struct too?

Reply:
AH yeah good catch. It seems I missed a couple fields.

Reply:
yeah it just occurred to me when I saw you were moving flakiness, there are a few not prefixed with `Test`.